Test task completed

KonstantinBel opened the pull request:
:up: @idbolshakov

KonstantinBel committed:
Common works complite

KonstantinBel committed:
Fixed IE11 and Edge problems

KonstantinBel committed:
Fixed codestyle

KonstantinBel committed:
Changed cart style

KonstantinBel committed:
Fixed deleting item from cart

KonstantinBel committed:
Refact Listing class

KonstantinBel committed:
Merge branch 'master' of https://github.com/Bartzlo/apress-frontend_test_task

KonstantinBel committed:
Refacted and did full modular

idbolshakov commented:
Привет, все круто! 

Допустим, что заказчикам очень понравился результат, но они просят чуть-чуть доработать:

* юзабилити-дизайнер очень просит, чтобы на всех кнопках курсор менялся на`pointer`

* дизайнер интерфейсов хотел бы, чтобы при открытии попапа заказа остальная страница чуть чуть затемнялась

* специалисту по оптимизации не очень нравятся твои `setTimeout` для создания анимации и стилизация напрямую в js через `.style.SomeCssProperty`, вместо этого он предлагает 
использовать `transition` для анимации и стилизацию через задание классов, а не свойств

* также заказчики планируют использовать разработанные тобой компоненты не все сразу а по отдельности на разных страницах. Поэтому просят предусмотреть возможно подключать каждый компонент (листинг, попап заказа, попап корзины) отдельно

* также файловая структура проекта, на котором будут использоваться твои компоненты предполагает, что все скрипты и стили находятся в папке `assets`

KonstantinBel committed:
Divide moduls into separate files (js, css)
Add pinter cursor for each buttons
Add detailed discription to JSDoc's
Add separate module usage examples into main.js
Move code to assets folder
Move event listeners into separate private functions
Move templates from index.html into module classes
Remove all element.style.*prop*
Remove excess setTimeOut's
Change orderPopup mask color

KonstantinBel commented:
🆙 @idbolshakov
Все доработал.
Модули разделил, теперь их можно без проблем использовать по отдельности (примеры в main.js).

Поясню про модуль Listener. Для того что бы его возможно было использовать отдельно, добавил возможность передать в конструкторе функции вместо объектов orderPopup и cartPopup. Эти функции будут вызваны при нажатии на кнопки "Заказать" и "В корзину" и получат данные соответствующего товара. Так же есть возможность передать свои объекты если они реализуют определенные методы (подробности указаны в JSDoc к классу Listener)

KonstantinBel committed:
Remove creating orderPopup and cartPoput ftom Listing

Это необходимо для того что бы клиенстский код мог сам решать где разместить элементы с попапами и убрать необходимость реализовать метод getElement() у объектов которые могут быть переданы вместо нативных CartPopup и OrderPopup
